testing/MLDB-2180-dataset-split.py

```diff
@@ -0,0 +1,184 @@
+        n = mldb.get('/v1/query', q="SELECT count(*) FROM ds_test", format='atom').json()
+        self.assertEqual(4, n)
+
+        res1 = mldb.query("SELECT sum({*}) FROM ds_train")
+        res2 = mldb.query("SELECT sum({*}) FROM ds_test")
+
+        self.assertEquals(res1, [["_rowName", "sum({*}).x", "sum({*}).y"],
+                                 ["[]", 13, 3 ]])
+
+        self.assertEquals(res2, [["_rowName", "sum({*}).x", "sum({*}).y"],
+                                 ["[]", 3, 1 ]])
+
+        mldb.put("/v1/procedures/split", {
+            "type": "split.train",
+            "params": {
+                "labels": "SELECT * FROM ds3",
+                "splits": [0.8, 0.2],
+                "outputDatasets": [{ "id": "ds_train",
+                                   "type": "sparse.mutable" },
+                                   { "id": "ds_test",
+                                   "type": "sparse.mutable" }],
+            }
+        })   
+
+        n = mldb.get('/v1/query', q="SELECT count(*) FROM ds_train", format='atom').json()
+        self.assertEqual(16, n)
+        n = mldb.get('/v1/query', q="SELECT count(*) FROM ds_test", format='atom').json()
+        self.assertEqual(4, n)
+
+        res1 = mldb.query("SELECT sum({*}) FROM ds_train")
+        res2 = mldb.query("SELECT sum({*}) FROM ds_test")
+
+        self.assertEquals(res1, [["_rowName", "sum({*}).x", "sum({*}).y"],
+                                 ["[]", 8, 8 ]])
+
+        self.assertEquals(res2, [["_rowName", "sum({*}).x", "sum({*}).y"],
+                                 ["[]", 2, 2 ]])
+
+    def test_testintersection(self):
+
+        mldb.put("/v1/procedures/split", {
+            "type": "split.train",
+            "params": {
+                "labels": "SELECT * FROM ds4",
+                "splits": [0.8, 0.2],
+                "foldImportance" : 1.0,
+                "outputDatasets": [{ "id": "ds_train",
+                                   "type": "sparse.mutable" },
+                                   { "id": "ds_test",
+                                   "type": "sparse.mutable" }],
+            }
+        })   
+
+        n = mldb.get('/v1/query', q="SELECT count(*) FROM ds_train", format='atom').json()
+        self.assertEqual(18, n)
+        n = mldb.get('/v1/query', q="SELECT count(*) FROM ds_test", format='atom').json()
+        self.assertEqual(6, n)
+
+        res1 = mldb.query("SELECT sum({*}) FROM ds_train")
+        res2 = mldb.query("SELECT sum({*}) FROM ds_test")
+
+        self.assertEquals(res1, [["_rowName", "sum({*}).x", "sum({*}).y", "sum({*}).z"],
+                                 ["[]", 8, 12, 7 ]])
+
+        self.assertEquals(res2, [["_rowName", "sum({*}).x", "sum({*}).y", "sum({*}).z"],
+                                 ["[]", 2, 3, 2 ]])
+
+        mldb.put("/v1/procedures/split", {
+            "type": "split.train",
+            "params": {
+                "labels": "SELECT * FROM ds4",
+                "splits": [0.8, 0.2],
```

Comment:
One a use case we have to with three datasets split.  I would add a test for that too.